text/0000-serialization.md

```diff
@@ -0,0 +1,187 @@
+      "content": 123
+    }
+  ]
+}
+```
+
+### Refracted Metadata & Attributes
+
+If the array's title from the example above needs to contain extra information, for example an encoding, then it can be refracted:
+
+```js
+{
+  "element": "array",
+  "meta": {
+    "title": {
+      "element": "string",
+      "attributes": {
+        "encoding": "utf-8"
+      },
+      "content": "My array"
+    }
+  },
+  "content": [
+    {
+      "element": "string",
+      "content": "Hello, world!"
+    },
+    {
+      "element": "number",
+      "content": 123
+    }
+  ]
+}
+```
+
+## Compact (list-based) JSON
+
+The short-form of the full JSON described above is based on lists. Each element is represented by a four item list in which the position of the item determines its semantic meaning. Each list consists of [`element`, `meta`, `attributes`, `content`].
+
+- `element`: The name of the element, e.g. `'string'`
+- `meta`: An object containing element metadata
+- `attributes`: An object containing element attributes
+- `content`: The element's content, which can be any type.
+
+### Example
+
+The following example describes an array named `My array` which looks like `['Hello, world!', 123]`:
+
+```js
+["array", {"title": "My array"}, {}, [
+  ["string", {}, {}, "Hello, world!"],
+  ["number", {}, {}, 123]
+]]
+```
+
+### Refracted Metadata & Attributes
+
+If the array's title from the example above needs to contain extra information, for example an encoding, then it can be refracted:
+
+```js
+["array", {
+    "title": ["string", {}, {"encoding": "utf-8"}, "My array"]
+  }, {}, [
+  ["string", {}, {}, "Hello, world!"],
+  ["number", {}, {}, 123]
+]]
+```
+
+## Preventing Ambiguity
+
+Because an element's metadata and attributes can contain both refracted elements and any other data, as well as fields which may be one or the other depending on content, we must prevent ambiguity. This can be accomplished by **always refracting ambiguous cases**. For example, given this **raw attribute** data:
+
+```js
+// Ambiguous raw content for full JSON serialization
+{
+  "element": "string",
+  "content": "foo"
+}
+
+// Ambiguous raw content for compact JSON serialization
+['foo', {}, {}, 0]
+```
+
+It must be serialized so that the parser knows it is **not an element**:
+
+```js
+// Unambiguous full JSON version
+{
+  "element": "object",
+  "content": [
+    {
+      "element": "member",
+      "content": {
+        "key": {
+          "element": "string",
+          "content": "element"
+        },
+        "value": {
+          "element": "string",
+          "content": "string"
+        }
+      }
+    },
+    {
+      "element": "member",
+      "content": {
+        "key": {
+          "element": "string",
+          "content": "content"
+        },
+        "value": {
+          "element": "string",
+          "content": "foo"
+        }
+      }
+    }
+  ]
+}
+
+// Unambiguous compact JSON version
+['array', {}, {}, ['foo', {}, {}, 0]]
```

Comment:
A couple of thoughts on these examples.

The full version is simply refracting the element as a normal object. I believe we would do this anyway within Minim, and would treat it as a normal Object Element. 

Regarding the compact example, I don't believe this is a valid array element. I believe it would have to be:

```js
['array', {}, {}, [
  ['string', {}, {}, 'foo'],
  ['object', {}, {}, []],
  ['object', {}, {}, []],
  ['number', {}, {}, 0]
]]
```

We would still do this in any situations where Compact Refract was not expected I believe.

So I actually believe, without any domain-specific knowledge, we will actually already do as you've defined. The difficulty is knowing where that domain-specific knowledge is in the tree.